pkg/reader/rmn_home.go

```diff
@@ -69,12 +75,80 @@ type rmnHomePoller struct {
 	pollingDuration      time.Duration // How frequently the poller fetches the chain configs
 }
 
-func NewRMNHomePoller(
+// GetRMNHomePoller returns a rmnHomePoller instance if it already exists, else creates a new one.
+// Returned rmnHomePoller is always started, therefore user of that API doesn't need to care about
+// concurrency and accidentally starting the same poller multiple times.
+//
+// In most of the cases, there is going to be only a single RMNHome deployed on a single chain. However,
+// OCR3Config allows to pass different RMNHome addresses for different chains. Therefore, we need to
+// support that and maintain a separate rmnHomePoller for each RMNHome address.
+// Having a singleton here is aimed to reduce the background polling to the underlying RPC node.
+func GetRMNHomePoller(
+	ctx context.Context,
+	lggr logger.Logger,
+	rmnHomeChainSelector cciptypes.ChainSelector,
+	rmnHomeAddress []byte,
+	contractReader contractreader.ContractReaderFacade,
+	pollingInterval time.Duration,
+) (RMNHome, error) {
+	instancesMu.Lock()
+	defer instancesMu.Unlock()
+
+	hexEncodedAddr := " 0x" + hex.EncodeToString(rmnHomeAddress)
+	key := fmt.Sprintf("%s-%s", rmnHomeChainSelector.String(), hexEncodedAddr)
+
+	instance, ok := getInstance(key)
+	if ok {
+		lggr.Infow("RMNHomePoller already exists, reusing instance",
+			"chainSelector", rmnHomeChainSelector,
+			"address", hexEncodedAddr,
+		)
+		return instance, nil
+	}
+
+	rmnHomeBoundContract := types.BoundContract{
+		Address: hexEncodedAddr,
+		Name:    consts.ContractNameRMNHome,
+	}
+
+	if err := contractReader.Bind(ctx, []types.BoundContract{rmnHomeBoundContract}); err != nil {
+		return nil, fmt.Errorf("failed to bind RMNHome contract: %w", err)
+	}
+
+	rmnHomeReader := newRMNHomePoller(
+		contractReader,
+		rmnHomeBoundContract,
+		logutil.WithComponent(lggr, "RMNHomePoller"),
+		pollingInterval,
+	)
+
+	if err := rmnHomeReader.Start(ctx); err != nil {
```

Comment:
How will closing happen? I think right now every plugin is closing their own RMN home poller instance

Reply:
A sort of dirty approach is to use https://pkg.go.dev/runtime#SetFinalizer if we don't have clear ownership semantics - otherwise maybe in the chainlink node delegate?

Reply:
Oh shoot, we need some sort of counting references here. Therefore close only when every commit holding reference is closed

Reply:
Btw, the bug is already there, because we never close `rmnHomeReader`

Reply:
Re: counting references thats what `SetFinalizer` does, might be less error prone